src/SimpleSoft.Database.NH/NHServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Linq;
+using SimpleSoft.Database;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extensions for <see cref="IServiceCollection"/> instances.
+    /// </summary>
+    public static class NHServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Registers into the container wrappers for each context operation, enabling
+        /// the usage of interfaces like <see cref="IReadByExternalId{TEntity}"/>.
+        /// </summary>
+        /// <remarks>
+        /// Services are added using variations of `TryAdd...`, ensuring you can swap
+        /// particular interface implementations by your own either by adding before or after
+        /// using the <see cref="ServiceCollectionDescriptorExtensions.Replace"/> method.
+        /// </remarks>
+        /// <param name="services"></param>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException" />
+        public static IServiceCollection AddDbContextOperations(
```

Comment:
Rename method to `AddSessionOperations` to be consistent to NHibernate terminology.